internal/queue/connect.go

```diff
@@ -0,0 +1,26 @@
+package queue
+
+import (
+	"context"
+	"fmt"
+	"net/url"
+
+	"github.com/JakeHumphries/gymshark-hacker-news/internal/models"
+	log "github.com/sirupsen/logrus"
+	"github.com/streadway/amqp"
+)
+
+func connect(ctx context.Context, cfg models.Config) (*amqp.Connection, error) {
```

Comment:
I would be inclined to have a max number of retries or timeout on this. You could also use a context with timeout too